src/Association/Embedded.php

```diff
@@ -131,34 +159,98 @@ public function entityClass($name = null)
     }
 
     /**
-     * Get/set the index class used for this embed.
+     * Sets the entity/document class used for this embed.
      *
-     * @param string|null|\Cake\ElasticSearch\Index $name The class name to set.
+     * @param string $name The name of the class to use
+     * @return $this
+     */
+    public function setEntityClass($name)
```

Comment:
There is one minor change left on your other pull request. After that I can get it merged. Once it is merged, could these changes be rebased as there are a lot of overlapping changes right now.

Reply:
I tried review boths PR and don't find anything left - just some docs. Can you point to me?

Updated with docs and a fix for Index looking for documents on entity namespace. Included some tests.